### PR TITLE
[MAINTENANCE] Minor clean up in style of an RBP test fixture; making variables access more robust

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-497/GREAT-596
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/rule_based_profiler.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def reconcile_profiler_variables(
         :return: reconciled variables in their canonical ParameterContainer object form
         """
         effective_variables: ParameterContainer
-        if variables is not None and isinstance(variables, dict):
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/rule_based_profiler.py:384]: Why this change? Won't variables either be a populated dict or `None`? If so, let's be explicit in our check for nullability.

File: great_expectations/validator/validator.py [hunk 2]
```diff
@@ -766,6 +766,7 @@ def resolve_validation_graph(
         graph: ValidationGraph,
         metrics: Dict[Tuple[str, str, str], Any],
         runtime_configuration: Optional[dict] = None,
+        min_graph_edges_pbar_enable: int = 0,  # Set to low number (e.g., 3) to suppress progress bar for small graphs.
```
Comment (cdkini) [inline great_expectations/validator/validator.py:769]: Nitpick - could we possibly rename this? Maybe something like `progress_bar_edge_threshold`? I'm not the best with naming so feel free to keep it as is if you'd like 😄

File: great_expectations/validator/validator.py [hunk 3]
```diff
@@ -812,7 +813,7 @@ def resolve_validation_graph(
                     if "metric_calculations" in progress_bars:
                         disable = not progress_bars["metric_calculations"]
 
-            if len(graph.edges) < 3:
```
Comment (cdkini) [inline great_expectations/validator/validator.py:815]: Any reason why we don't set the default to 3? This will change behavior outside of the RBP usecase. I think an argument can be made to keep this value the same.

**Review by cdkini — COMMENTED** [review]
Very minor comments but nothing really blocking. Please let me know your thoughts and I will approve 🎉

File: tests/test_fixtures/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml [hunk 4]
```diff
@@ -126,7 +126,7 @@ rules:
         module_name: great_expectations.rule_based_profiler.expectation_configuration_builder
         column: $domain.domain_kwargs.column
         min_value: $variables.min_timestamp
-        max_value: $parameter.my_max_event_ts.value[0]
+        max_value: $parameter.my_max_event_ts.value[-1]
```
Comment (NathanFarmer) [inline tests/test_fixtures/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml:129]: ❤️ I will keep this pattern in mind...

File: great_expectations/validator/validator.py (hunk 2 above)
Comment (NathanFarmer) [reply great_expectations/validator/validator.py:769]: Why do we count the edges and not the nodes of the graph anyway?

**Review by NathanFarmer — COMMENTED** [review]
A few comments...

File: great_expectations/rule_based_profiler/rule_based_profiler.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/rule_based_profiler.py:384]: @cdkini When used from Expectations, sometimes `variables` is omitted entirely -- so we have to check for that, too.  Thanks!

File: great_expectations/validator/validator.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/validator/validator.py:769]: @cdkini I do not mind a better name; for now -- since the logic talks about edges, let us leave like this -- we can change later once a potentially better name is confirmed.  Thanks!

File: great_expectations/validator/validator.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/validator/validator.py:815]: @cdkini I was observing an inconsistency in notebooks, and metrics were not being reported, so one had to wonder if there is a bug.  With reporting them, it makes it easier, and I have not seen any material change in output amount.  Thanks.

File: tests/test_fixtures/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml (hunk 4 above)
Comment (alexsherstinsky) [reply tests/test_fixtures/rule_based_profiler/alice_user_workflow_verbose_profiler_config.yml:129]: @NathanFarmer  Yes -- the reason this is important, is because we want the "most recent batch".  Thanks! /cc @cdkini

**Comment by alexsherstinsky** [comment]
@NathanFarmer Regarding the edges -- you are correct, this points to the "home brew" nature of our graph evaluation algorithm.  I have a long-standing comment in that code in `Validator` to improve one aspect of it, and it would be better to overhaul it entirely. /cc @cdkini @NathanFarmer

**Review by cdkini — APPROVED** [review]
